apps/ejabberd/src/gen_mod_deps.erl

```diff
@@ -0,0 +1,233 @@
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(gen_mod_deps).
+
+-include("ejabberd.hrl").
+
+-type ejd_module_params() :: proplists:proplist().
```

Comment:
Do we need the `ejd_` prefix in these type definitions? If so, it'd be more accurate to use other prefix. `ejd` is a shortcut for `ejabberd` and this code is very MongooseIM specific.